frontend/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from "react"
+import { Route, Redirect, RouteProps } from "react-router-dom"
+import LoadingPage from "../FullPageRoutes/LoadingPage"
+import { AppContext, useStatefulLocation } from "../utils/context"
+
+type ProtectedRouteProps = RouteProps
+
+export function ProtectedRoute({
+  children,
+  ...rest
+}: ProtectedRouteProps): JSX.Element {
+  const {
+    user: [userValue],
+  } = React.useContext(AppContext)
+  const location = useStatefulLocation()
+
+  return (
```

Comment:
This might be asking for too much but we are currently still able to go to the sign in/sign up page when logged in. It would be probably be weird to be able to log in if already logged in... Although, I think what we have now is sufficient enough for the presentation.

Reply:
How did you do that?

Reply:
After I was logged in, I simply changed the route back to /tokens and it took me back to the login/sign up page

Reply:
Ah, I think that's fine